aimsprop/fms90.py

```diff
@@ -186,13 +186,13 @@ def _parse_spawnlog(filepath: Path, initial_I: int = None) -> Dict[Any, Any]:
     else:
         with open(Spawnfile) as f:
             lines = f.readlines()[1:]
-
-    for lind, line in enumerate(lines):
-        # match groups are TBF ID, state, parent TBF ID, parent TBF state
-        mobj = re.match(r"^\s*\S+\s+\S+\s+\S+\s+(\d+)\s+(\d+)\s+(\d+)\s+(\d+)", line)
-        states[int(mobj.group(1))] = int(mobj.group(2))
-        # Redundant, but captures IC TBFs
-        states[int(mobj.group(3))] = int(mobj.group(4))
+        # CB: if we have the Spawn.log, read its content
+        for lind, line in enumerate(lines):
+            # match groups are TBF ID, state, parent TBF ID, parent TBF state
+            mobj = re.match(r"^\s*\S+\s+\S+\s+\S+\s+(\d+)\s+(\d+)\s+(\d+)\s+(\d+)", line)
+            states[int(mobj.group(1))] = int(mobj.group(2))
+            # Redundant, but captures IC TBFs
+            states[int(mobj.group(3))] = int(mobj.group(4))
```

Comment:
What exactly does indenting these line do? These lines will execute if the spawn log is found without the indentation :) The Python is more idiomatic without the indentation as well. 

If the spawn log is not found the `RuntimeError` will be raised, otherwise the lines will be read and the for loop will execute.

Am I missing something?

Thanks for keeping an eye on the code :)

Reply:
I believe the issue is that not all trajectories actually have Spawn.log but you might still want to analyze them, yes?

Reply:
Yes, the issue is exactly that. "lines" cannot be read if there is no Spawn.log. However, with the current indentation, it is read regardless of whether there is a Spawn.log or not. In the latter case, it fails in the current default version.